src/Tribe/Views/V2/Views/List_View.php

```diff
@@ -98,19 +98,34 @@ protected function get_past_url( $canonical = false, $page = 1 ) {
 		$date           = $this->context->get( 'event_date', $default_date );
 		$event_date_var = $default_date === $date ? '' : $date;
 
+
 		$past = tribe_events()->by_args( $this->setup_repository_args( $this->context->alter( [
 			'event_display_mode' => 'past',
 			'paged'              => $page,
 		] ) ) );
 
 		if ( $past->count() > 0 ) {
-			$past_url_object = clone $this->url->add_query_args( array_filter( [
+
+			$query_args = [
 				'post_type'        => TEC::POSTTYPE,
 				'eventDisplay'     => 'past',
 				'eventDate'        => $event_date_var,
 				$this->page_key    => $page,
 				'tribe-bar-search' => $this->context->get( 'keyword' ),
-			] ) );
+			];
+
+			/**
+			 * Filters the query arguments that will be used to build a View URL.
+			 *
+			 * @since TBD
+			 *
+			 * @param array          $query_args An array of query args that will be used to build the URL for the View.
+			 * @param View_Interface $this       This View instance.
+			 * @param bool           $canonical  Whether the URL should be the canonical one or not.
+			 */
+			$query_args = apply_filters( 'tribe_events_views_v2_url_query_args', $query_args, $this, $canonical );
```

Comment:
Optional thing, just wanted to ask the question: in name and signature this is identical to what we see [down on line 200 (same file)](https://github.com/moderntribe/the-events-calendar/pull/2931/files#diff-a30e077e32723b54e7acf6fe03e4b685R200). Could it be worth moving it to its own method? That would keep things DRYer and, thinking of the potential for error during future maintenance, would reduce the potential for one to fall out of sync with the other.

Reply:
I've created a method in the View class to handle query_arg filtering and plugged it in to all the places! Good call :+1: